airbyte-cdk/python/unit_tests/sources/file_based/test_scenarios.py

```diff
@@ -271,7 +271,27 @@ def _verify_expected_logs(logs: List[Dict[str, Any]], expected_logs: Optional[Li
 
 @pytest.mark.parametrize("scenario", spec_scenarios, ids=[c.name for c in spec_scenarios])
 def test_spec(capsys: CaptureFixture[str], scenario: TestScenario) -> None:
-    assert spec(capsys, scenario) == scenario.expected_spec
+    assert compare_dicts_with_oneof(spec(capsys, scenario), scenario.expected_spec)
+
+
+def compare_dicts_with_oneof(dict1, dict2):
+    if not isinstance(dict1, dict) or not isinstance(dict2, dict):
+        return dict1 == dict2
+
+    if set(dict1.keys()) != set(dict2.keys()):
+        return False
+
+    for key in dict1:
+        if key == 'oneOf' and isinstance(dict1[key], list) and isinstance(dict2[key], list):
+            # If the key is 'oneOf', and the values are lists, then compare ignoring order
+            l1, l2 = sorted(dict1[key], key=lambda x: x["type"]), sorted(dict2[key], key=lambda x: x["type"])
```

Comment:
Can we have multiple entries of the same type? I can see a case where this comparison could pass even though it should fail. For example, expected could have (List[int] and List[str]) and actual could have (List[float] and List[str]). 

Can we cast the list as a set and compare the sets? This assumes that entries are hashable which I think is a fair assumption in this case